### ISSUE TITLE
forge script fails to verify on mumbai

### ISSUE THREAD

**Issue by thedavidmeister** [foundry-rs/foundry#5170]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [ ] Foundryup

### What command(s) is the bug in?

forge script

### Operating System

Linux

### Describe the bug

as seen here in CI https://github.com/rainprotocol/rain.orderbook/actions/runs/5291795497/jobs/9577824407

forge script deploys a contract but fails to verify it

```
We haven't found any matching bytecode for the following contracts: [0xe80e098a8379eee4c05100e36088fd13e9e7b9b3].

This may occur when resuming a verification, but the underlying source code or compiler version has changed.
```

This seems strange to me as the code should be built and deployed as-is by CI without any modifications or "resuming" a verification.

**Comment by Evalir** [comment]
Ref https://github.com/foundry-rs/foundry/issues/5168

**Comment by thedavidmeister** [comment]
it verifies now strangely even with the same error https://github.com/rainprotocol/rain.orderbook/actions/runs/5291881682/jobs/9578030758?pr=1

**Comment by Evalir** [comment]
Closing, we merged #5167 and will be available in the next nightly
